lib/upload.js

```diff
@@ -531,6 +534,40 @@ class BaseUpload {
       return
     }
 
+    if (this.options.enableChecksum) {
+      const optionsReq = this._openRequest('OPTIONS', this.options.endpoint)
+      this._sendRequest(optionsReq)
+        .then((res) => {
+          const isChecksumSupported = res
+            .getHeader('Tus-Extension')
+            .split(',')
+            .includes('checksum')
+          if (!isChecksumSupported) {
+            log('Checksum is not supported by this tus server')
+            return
+          }
+          const supportedAlgorithms = res.getHeader('Tus-Checksum-Algorithm')
+          if (!supportedAlgorithms) {
+            throw new Error(
+              'tus: checksum is enabled, but Tus-Checksum-Algorithm is not defined by tus server',
+            )
+          }
+          let Checksum
+          if (window) {
+            // eslint-disable-next-line global-require
+            Checksum = require('./browser/extensions/checksum').default
+          } else {
+            // eslint-disable-next-line global-require
+            Checksum = require('./browser/extensions/checksum').default
+          }
+          this._checksum = new Checksum(
+            Checksum.supportedAlogrithms.find((supportedAlgo) => supportedAlgorithms.includes(supportedAlgo)),
```

Comment:
What does this check do? The combination of find and includes looks suspicious.

Reply:
I was trying to fetch the supported algorithms from the server and see if one of them matches one of the supported  algorithms on the client. If we're saying client and server have agreed upon an algorithm from the get go, then we can skip this OPTIONS call and let client just configure the algorithm in options.